hi-ml-cpath/src/health_cpath/models/deepmil.py

```diff
@@ -89,7 +89,9 @@ def __init__(self,
 
         # Model components
         self.encoder = encoder_params.get_encoder(outputs_folder)
-        self.aggregation_fn, self.num_pooling = pooling_params.get_pooling_layer(self.encoder.num_encoding)
+        self.projector = encoder_params.get_projection_layer(self.encoder.num_encoding)
+        num_encoding = encoder_params.projection_dim if encoder_params.projection_dim > 0 else self.encoder.num_encoding
```

Comment:
could you explain why this is needed in a comment, perhaps

Reply:
If the projection_dim is 0 than the projection layer is an Identity layer so num_encoding won't change, however, if encoder_params.projection_dim > 0, we reduce the encoding dimension so we need to use encoder_params.projection_dim

Reply:
could you please add it as a comment (what I implied above)